src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -595,6 +639,21 @@ private DataAccessRequest updateDarWithDocumentContents(
     if (!dar.getDraft()) {
       throw new SubmittedDARCannotBeEditedException();
     }
+    uploadDocumentContents(type, dar, uploadInputStream, fileDetail);
+    return dataAccessRequestService.updateByReferenceId(user, dar);
+  }
+
+  /**
+   * Uploads the document contents to GCS and mutates the dar data object with the blobId.
+   *
+   * @param type            The type of document (IRB or Collaboration)
+   * @param dar             The Data Access Request
+   * @param uploadInputStream The input stream of the file to be uploaded
+   * @param fileDetail      The file details
+   * @throws IOException if an error occurs during upload
+   */
+  public void uploadDocumentContents(DarDocumentType type, DataAccessRequest dar, InputStream uploadInputStream,
```

Comment:
I think the use of this method in this way introduces the chance a child DAR could delete the IRB and Collaboration documents on the parent DAR.  Is that the behavior you expect here?

Reply:
I see this too - on line 585, we copy the parent dar's values for `irbDocumentLocation` and `collaborationLetterLocation` into the child. The end result here will be that the parent dar will contain a reference to a location that will have had it's underlying file store removed and the child will contain new values with references to new documents.

Reply:
@rjohanek Good question on what to do here.
The thing crossing my mind at a high level is if we need to maintain a DAR exactly as it was submitted forever so that we can demonstrate what was approved at the time, if we need to do that, my suggestion is below:
1) If we have existing documents from a (parent) DAR, we copy the files (making new FileStorageObjects) to new locations with new blobIDs for them so they can live as independent artifacts for this DAR instance.
2) We accept new documents when posting a progress report and when we've got new documents, we only use those.

Might be a good follow-up question for our POs.

Reply:
@rjohanek - great conversation.  I really appreciate your approach of incremental progress.   We discussed the updated code path and some follow-on tickets you would make for our product owners to consider:
1) Should collaboration agreements and IRB documents be copied over from the parent DAR to this new DAR?
2) When we delete any DAR in the DAR collection, should we delete all DARs? (e.g. a cascade delete)

IMHO, the answers to the questions above don't need to block completion of this ticket because the new tickets give us an opportunity to revisit the decisions made and make changes as needed.